### PR TITLE
Create container for FormField message and allow custom icon

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
This PR allows users to specify icons that accompany the FormField error or info messages. Additionally, they are able to style the container that surrounds the icon and message. This PR introduces `formField.error.container`, `formField.info.container`, `formField.error.icon`, and `formField.info.icon` to the theme.

#### Where should the reviewer start?
src/js/components/FormField/FormField.js

#### What testing has been done on this PR?
Added a jest test and tested in Form storybook titled `Themed`.

#### How should this be manually tested?
Apply any of the new theme objects to a custom storybook.

#### What are the relevant issues?
Builds off of https://github.com/grommet/grommet/pull/4363
Closes https://github.com/grommet/grommet/issues/4362

#### Do the grommet docs need to be updated?
Yes, updated in this PR.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/FormField/FormField.js [hunk 1]
```diff
@@ -45,10 +45,41 @@ const FormFieldContentBox = styled(Box)`
...
+  ${props =>
+    props.messageType &&
+    props.theme.formField[props.messageType].container &&
+    props.theme.formField[props.messageType].container.extend}
+`;
+
+const Message = ({ error, info, message, ...rest }) => {
+  const theme = useContext(ThemeContext) || defaultProps.theme;
+
+  let messageType;
```
Comment (ericsoderberghp) [inline src/js/components/FormField/FormField.js:58]: Lines 58-66 could be inside `if (message) {`

File: src/js/components/FormField/FormField.js [hunk 2]
```diff
@@ -45,10 +45,41 @@ const FormFieldContentBox = styled(Box)`
...
-    if (typeof message === 'string') return <Text {...rest}>{message}</Text>;
-    return <Box {...rest}>{message}</Box>;
+    let messageContent;
+    if (typeof message === 'string')
+      messageContent = <Text {...rest}>{message}</Text>;
+    else messageContent = <Box {...rest}>{message}</Box>;
+
+    return (
+      <StyledMessageContainer
+        direction={icon ? 'row' : 'column'}
```
Comment (ericsoderberghp) [inline src/js/components/FormField/FormField.js:75]: Why not always `direction="row"`?

Comment (taysea) [reply src/js/components/FormField/FormField.js:75]: You're right, I think always having row is fine.

File: src/js/components/FormField/doc.js [hunk 3]
```diff
@@ -163,6 +163,28 @@ export const themeDoc = {
...
     defaultValue: 'status-critical',
   },
+  'formField.error.container': {
+    description: `Any valid Box props for the container surrounding the error 
+    message and icon.`,
+    type: 'object',
+    defaultValue: undefined,
+  },
+  'formField.error.icon': {
+    description: 'An icon placed next to the error message.',
```
Comment (ShimiSun) [inline src/js/components/FormField/doc.js:173]: Can you please explain better what 'next' mean.

File: src/js/components/FormField/FormField.js [hunk 4]
```diff
@@ -45,10 +45,41 @@ const FormFieldContentBox = styled(Box)`
...
+    props.theme.formField[props.messageType].container.extend}
+`;
+
+const Message = ({ error, info, message, ...rest }) => {
+  const theme = useContext(ThemeContext) || defaultProps.theme;
+
   if (message) {
-    if (typeof message === 'string') return <Text {...rest}>{message}</Text>;
-    return <Box {...rest}>{message}</Box>;
+    let messageType;
```
Comment (taysea) [inline src/js/components/FormField/FormField.js:59]: check what happens if messageType is undefined

File: src/js/components/FormField/FormField.js [hunk 5]
```diff
@@ -45,10 +45,41 @@ const FormFieldContentBox = styled(Box)`
...
+      theme.formField[messageType] && theme.formField[messageType].icon;
+
+    const containerProps =
+      theme.formField[messageType] && theme.formField[messageType].container;
+
+    let messageContent;
+    if (typeof message === 'string')
+      messageContent = <Text {...rest}>{message}</Text>;
+    else messageContent = <Box {...rest}>{message}</Box>;
+
```
Comment (taysea) [inline src/js/components/FormField/FormField.js:72]: if icon isn't defined, don't wrap in messagecontainer

File: src/js/components/FormField/FormField.js [hunk 6]
```diff
@@ -356,8 +387,8 @@ const FormField = forwardRef(
           undefined
         )}
         {contents}
-        <Message message={error} {...formFieldTheme.error} />
-        <Message message={info} {...formFieldTheme.info} />
+        <Message error message={error} {...formFieldTheme.error} />
```
Comment (taysea) [inline src/js/components/FormField/FormField.js:390]: change to `type`

File: src/js/components/FormField/stories/Themed.js [hunk 7]
```diff
@@ -56,7 +76,8 @@ const Themed = () => {
               name="example1"
               label="Field Label"
               help="Some helpful descriptive text"
-              error="Message to show on error"
+              error={`Message to show on error. This is a long message to 
+              demonstrate custom svg alignment.`}
               info="Additional contextual information"
```
Comment (taysea) [inline src/js/components/FormField/stories/Themed.js:81]: Make this message longer to show align center

**Pull request merged**